src/Nethermind/Nethermind.Blockchain/FullPruning/CopyTreeVisitor.cs

```diff
@@ -62,7 +65,7 @@ public void VisitMissingNode(Hash256 nodeHash, TrieVisitContext trieVisitContext
             }
 
             // if nodes are missing then state trie is not valid and we need to stop copying it
-            _pruningContext.CancellationTokenSource.Cancel();
+            throw new TrieException($"Trie {nodeHash} missing");
```

Comment:
add to method:
    [DoesNotReturn]
    [StackTraceHidden]

Reply:
Done